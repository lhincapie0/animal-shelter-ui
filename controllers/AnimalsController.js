import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animals`,
            baseURL: process.env.API,
            headers: {
                'Content-Type': 'application/json',
            },
            data: animal,
            method: 'POST'
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(animalName) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${animalName}`
        });
    }
}

