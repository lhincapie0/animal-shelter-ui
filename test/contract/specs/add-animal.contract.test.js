import { getProvider } from './init-pact';
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const provider = getProvider();
describe('Given An Animal Service', () => {
    const animalMock = {
        name: "Manchitas",
        breed: "Crillo",
        gender: "Female",
        vaccinated:true,
    }

    beforeAll(async ()=>{
        await provider.setup();
    });

    describe('When a request the animals create', () =>{
        beforeAll(async ()=>{
            await provider.addInteraction({
                uponReceiving: 'a request to get the animal created',
                state:"create animal",
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    body: animalMock,
                },
                willRespondWith: {
                    status:201,
                    body: Matchers.like(animalMock),
                }
            });
        });
        it("Then it should return the right data", async() => {
            const response = await AnimalController.register(animalMock);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });
    });
    afterAll(async ()=>{
        await provider.finalize();
    });
});
