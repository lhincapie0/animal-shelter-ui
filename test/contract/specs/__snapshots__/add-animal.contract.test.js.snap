// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal Service When a request the animals create Then it should return the right data 1`] = `
Object {
  "breed": "Crillo",
  "gender": "Female",
  "name": "Manchitas",
  "vaccinated": true,
}
`;
